ldap/servers/slapd/add.c

```diff
@@ -523,30 +543,57 @@ op_shared_add(Slapi_PBlock *pb)
     proxy_err = proxyauth_get_dn(pb, &proxydn, &errtext);
 
     if (operation_is_flag_set(operation, OP_FLAG_ACTION_LOG_ACCESS)) {
+        slapd_log_pblock logpb = {0};
+
         if (proxydn) {
             proxystr = slapi_ch_smprintf(" authzid=\"%s\"", proxydn);
         }
 
+        logpb.log_format = log_format;
+        logpb.target_dn = slapi_entry_get_dn_const(e);
+        logpb.request_controls = operation_get_req_controls(operation);
+        logpb.authzid = proxydn;
+
         if (!internal_op) {
-            slapi_log_access(LDAP_DEBUG_STATS, "conn=%" PRIu64 " op=%d ADD dn=\"%s\"%s\n",
-                             pb_conn ? pb_conn->c_connid : -1,
-                             operation->o_opid,
-                             slapi_entry_get_dn_const(e),
-                             proxystr ? proxystr : "");
+            if (log_format != LOG_FORMAT_DEFAULT) {
+                /* JSON logging */
+                logpb.conn_time = pb_conn ? pb_conn->c_starttime : operation->o_conn_starttime;
```

Comment:
I think that we should have a function to initialize the logpb that  that set all fields needed for most of the operations 
void init_logpb(slapd_log_pblock *logpb, slapd_pblock *pb, int is_internal_op)
That will ensure that we are not forgetting initializing some fields like the sid and that will make the code shorter

Reply:
Also the pblock is not always available every time we make a log call.  Sometimes the conn and op structs are passed as arguments to the function that is making the logging call.  I'm just not sure we can consistently use a meaningful/useful init function as we still need to set unique struct elements for each logging call.  Sure, I can do:

    logpb_init(&logpb, start_time, conn_id, op_id, internal_op, inernal_nested_count)

But we will still have a bunch of other initializations to do for each logging call.  Thought it would be simpler and easier to read if it's all just done in the one spot.

Reply:
Really ? Since the pblock is used for most of the  operations / internal operations I suspect that in most case, we should have it (In fact, I do not know when we do not have it  but  for the few cases were it is not there we can always initializing all fields specifically as of today).
And the init fonctions should only initialize the parameter that are used in all operations
 isome connection/operation data,
  including incomming controls,
   sid,
   staring time ...
The operation specific one should be initialized on per case basic 
i.e: 
    logpb_init(&logpb, pb);
    logpb.bind_dn = dn;
    logpb.method = "sasl";

"simpler and easier to read" is arguable,  but probably not so easy to maintain ...
if we have to change something (like adding a parameter)  it will again impact a lot of code in different  files
having a generic initialization will ensure that the change occurs in a few places

Reply:
Ok I'll work on this and see how it unfolds :-)

Reply:
@progier389 ok changes made, and the CI failures are not reproducible on my side.